datasets/yelp2018.py

```diff
@@ -0,0 +1,89 @@
+from torchtext.data.iterator import BucketIterator
+from torchtext.vocab import Vectors
+
+
+def char_quantize(string, max_length=1000):
+    identity = np.identity(len(Yelp2018CharQuantized.ALPHABET))
+    quantized_string = np.array([identity[Yelp2018CharQuantized.ALPHABET[char]] for char in list(string.lower()) if char in Yelp2018CharQuantized.ALPHABET], dtype=np.float32)
+    if len(quantized_string) > max_length:
+        return quantized_string[:max_length]
+    else:
+        return np.concatenate((quantized_string, np.zeros((max_length - len(quantized_string), len(Yelp2018CharQuantized.ALPHABET)), dtype=np.float32)))
+
+
+def process_labels(string):
+    """
+    Returns the label string as a list of integers
+    :param string:
+    :return:
+    """
+    return [float(x) for x in string]
+
+
+class Yelp2018(TabularDataset):
+    NAME = 'Yelp2018'
+    NUM_CLASSES = 10
+    TEXT_FIELD = Field(batch_first=True, tokenize=clean_string, include_lengths=True)
+    LABEL_FIELD = Field(sequential=False, use_vocab=False, batch_first=True, preprocessing=process_labels)
+
+    @staticmethod
+    def sort_key(ex):
+        return len(ex.text)
+
+    @classmethod
+    def splits(cls, path, train=os.path.join('Yelp-Reviews-2018', 'data', 'yelp2018_train.tsv'),
+               validation=os.path.join('Yelp-Reviews-2018', 'data', 'yelp2018_validation.tsv'),
+               test=os.path.join('Yelp-Reviews-2018', 'data', 'yelp2018_test.tsv'), **kwargs):
+        return super(Yelp2018, cls).splits(
+            path, train=train, validation=validation, test=test,
+            format='tsv', fields=[('label', cls.LABEL_FIELD), ('text', cls.TEXT_FIELD)]
+        )
+
+    @classmethod
+    def iters(cls, path, vectors_name, vectors_cache, batch_size=64, shuffle=True, device=0, vectors=None,
+              unk_init=torch.Tensor.zero_):
+        """
+        :param path: directory containing train, test, dev files
+        :param vectors_name: name of word vectors file
+        :param vectors_cache: path to directory containing word vectors file
+        :param batch_size: batch size
+        :param device: GPU device
+        :param vectors: custom vectors - either predefined torchtext vectors or your own custom Vector classes
+        :param unk_init: function used to generate vector for OOV words
+        :return:
+        """
+        if vectors is None:
+            vectors = Vectors(name=vectors_name, cache=vectors_cache, unk_init=unk_init)
+
+        train, val, test = cls.splits(path)
+        cls.TEXT_FIELD.build_vocab(train, val, test, vectors=vectors)
+        return BucketIterator.splits((train, val, test), batch_size=batch_size, repeat=False, shuffle=shuffle,
+                                     sort_within_batch=True, device=device)
+
+
+class Yelp2018CharQuantized(Yelp2018):
+    ALPHABET = dict(map(lambda t: (t[1], t[0]), enumerate(list("""abcdefghijklmnopqrstuvwxyz0123456789,;.!?:'\"/\\|_@#$%^&*~`+-=<>()[]{}"""))))
```

Comment:
`list(...)` is unnecessary inside of `enumerate`, as are the triple quotes for the string itself (since it isn't really being used as a multi-line string). Just removing those would make this line a bit shorter but you could also break it into a couple lines to improve readability even more.

```python
In[2]: ALPHABET = dict(map(lambda t: (t[1], t[0]), enumerate(list("""abcdefghijklmnopqrstuvwxyz0123456789,;.!?:'\"/\\|_@#$%^&*~`+-=<>()[]{}"""))))
  ...: ALPHABE1 = {v: i for i, v in enumerate("abcdefghijklmnopqrstuvwxyz0123456789,;.!?:'\"/\\|_@#$%^&*~`+-=<>()[]{}")}
  ...: ALPHABE2 = {v: i for i, v in enumerate(
  ...:     "abcdefghijklmnopqrstuvwxyz0123456789,;.!?:'\"/\\|_@#$%^&*~`+-=<>()[]{}"
  ...: )}
In[3]: ALPHABET == ALPHABE1 == ALPHABE2
Out[3]: True
```